modulesAndFiles/tests.c

```diff
@@ -0,0 +1,50 @@
+#include "tests.h"
+#include "sortingAndSearching.h"
+#include "readingFromFile.h"
+#include <stdio.h>
+#include <stdbool.h>
+
+bool runTests(int array[], int size, int expectedResult) {
+    int result = findMostFrequent(array, size);
+    if (result == expectedResult) {
+        return true;
+    } else {
+        return false;
+	}
+}
+
+bool testForReading() {
+    int testArray[9] = { 0 };
+    readerForTest(testArray);
+    int expected = 1;
+    return (findMostFrequent(testArray, 9) == expected);
```

Comment:
```suggestion
    return findMostFrequent(testArray, 9) == expected;
```
Хотя я без идей, зачем Вам expected